theseus/optimizer/nonlinear/nonlinear_optimizer.py

```diff
@@ -16,6 +17,30 @@
 from theseus.optimizer import Optimizer, OptimizerInfo
 
 
+class BackwardMode(Enum):
```

Comment:
I was thinking all the way up to `Optimizer` but maybe here is okay for now?

Reply:
Let's leave here for now, as this is not currently relevant to any other optimizers. It'll be easy to move around if we ever need it a level up.